packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Expressions/SwitchCaseWhenExpression.cs

```diff
@@ -1,15 +1,17 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using Microsoft.Generator.CSharp.Snippets;
+
 namespace Microsoft.Generator.CSharp.Expressions
 {
-    public sealed record SwitchCaseWhenExpression(ValueExpression Case, BoolExpression Condition) : ValueExpression
+    public sealed record SwitchCaseWhenExpression(ValueExpression Case, BoolSnippet Condition) : ValueExpression
```

Comment:
My suggestion is that
if something accepts an input, the type should just be the widest `ValueExpression`.
if something produces an output, its type could be a snippet with type information.
For example, here we require this is a `BoolSnippet`, some one may write `v.Property("IsSomething")` which calls the property, and they should know the type of the property is a boolean, but here if they want to use that as a condition here, they will have to do:
```
new SwitchCaseWhenExpression(case, new BoolSnippet(v.Property("IsSomething"))
```
which does nothing about validation but introduces inconvenience. I think the purpose of snippets should introduce convenience instead of providing validations because it is impossible

And when we write something in the snippets, such as in `IntSnippet`, we write a property `MaxValue` which we know the type of this is `int`, therefore we put `IntSnippet` as its type, so that we could then call `Add` or `Minus` on it to produce convenience.

Reply:
Ok, I switched all the inputs for expressions and statements.